File: wrench-prefs-no-op/src/main/java/com/izettle/wrench/preferences/WrenchPreferences.java
package com.izettle.wrench.preferences;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;


public class WrenchPreferences {
    public WrenchPreferences(@NonNull Context context) {
    }

    @NonNull
    public <T extends Enum<T>> T getEnum(@NonNull String key, @NonNull Class<T> type, @NonNull T defValue) {
        return defValue;
    }

    @Nullable
    public String getString(@NonNull String key, @Nullable String defValue) {
        return defValue;
    }

    public boolean getBoolean(@NonNull String key, boolean defValue) {
        return defValue;
    }

    public int getInt(@NonNull String key, int defValue) {
        return defValue;
    }
}

File: wrench-prefs-no-op/src/main/java/com/izettle/wrench/preferences/WrenchPreferences.kt
package com.izettle.wrench.preferences


import android.content.Context

@Suppress("unused", "UNUSED_PARAMETER")
class WrenchPreferences(context: Context) {

    fun <T : Enum<T>> getEnum(key: String, type: Class<T>, defValue: T): T {
        return defValue
    }

    fun getString(key: String, defValue: String): String {
        return defValue
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        return defValue
    }

    fun getInt(key: String, defValue: Int): Int {
        return defValue
    }
}
